Added age, stance, height in cm, reach in cm and record to scraper.

DanielvanderZee opened the pull request.

DanielvanderZee committed:
Added age, stance, height in cm, reach in cm and record to scraper.

tomrijntjes commented:
A small semantic issue: the clean_rating method is reused for purposes not related to the fighter rating. I'd recommend renaming to extract_alphanumeric, or creating a seperate method. 

Will test later.

FlorisHoogenboom commented on boxrec/parsers.py:
```diff
@@ -78,6 +79,84 @@ def get_rating_after_fight(self, tree):
+                FightParser.BASE_DOM_PATH + \
+                '[./td/b/text() = "age"]/td[position() = 1 or position() =3]'
+        )
+        age_left = self.clean_rating(age_row[0].text)
```
Make sure to handle errors here.

FlorisHoogenboom commented on boxrec/parsers.py:
```diff
@@ -78,6 +79,84 @@ def get_rating_after_fight(self, tree):
+                FightParser.BASE_DOM_PATH + \
+                '[./td/b/text() = "stance"]/td[position() = 1 or position() =3]'
+        )
+        stance_left = stance_row[0].text.rstrip()
```
Also here, raise FailedToParse in case of failiures.

FlorisHoogenboom commented on boxrec/parsers.py:
```diff
@@ -78,6 +79,84 @@ def get_rating_after_fight(self, tree):
+                FightParser.BASE_DOM_PATH + \
+                '[./td/b/text() = "height"]/td[position() = 1 or position() =3]'
+        )
+        height_left = int(re.findall('\d+',height_row[0].text.split('/')[1].strip())[0])
```
Idem.

FlorisHoogenboom commented on boxrec/parsers.py:
```diff
@@ -78,6 +79,84 @@ def get_rating_after_fight(self, tree):
+                FightParser.BASE_DOM_PATH + \
+                '[./td/b/text() = "reach"]/td[position() = 1 or position() =3]'
+        )
+        reach_left = int(re.findall('\d+',reach_row[0].text.split('/')[1].strip())[0])
```
idem.

FlorisHoogenboom commented on boxrec/parsers.py:
```diff
@@ -78,6 +79,84 @@ def get_rating_after_fight(self, tree):
+                FightParser.BASE_DOM_PATH + \
+                '[./td/b/text() = "won"]/td[position() = 1 or position() =3]'
+        )
+        win_left = self.clean_rating(win_row[0].text)
```
Idem.

FlorisHoogenboom commented on boxrec/parsers.py:
```diff
@@ -78,6 +79,84 @@ def get_rating_after_fight(self, tree):
+        )
+        drawn_left = self.clean_rating(drawn_row[0].text)
+        drawn_right = self.clean_rating(drawn_row[1].text)
+        
```
Too much whitespace.

FlorisHoogenboom commented on boxrec/parsers.py:
```diff
@@ -78,6 +79,84 @@ def get_rating_after_fight(self, tree):
+    
+    def get_age(self,tree):
+        """New function to retreive age of boxer.
+        @Author daniel.van.der.zee."""
```
Please remove @Author directives from comments.

FlorisHoogenboom reviewed (changes requested):
Really nice pull request @DanielvanderZee! Great work!

There are some parts which raise errors during testing. These errors are not a consequence of your code, but rather the malformatting of the BoxRec pages. Could you make sure that everywhere where your PR might raise an error, these Errors are handled. In a situation in which you cannot recover from an error, make sure to raise a FailedToParse error with a meaningfull message. This way the end user can focus on handle these errors.

DanielvanderZee committed:
Updated parser according to comments by Floris Hoogenboom. Parsers now return None if values are not included on Boxrec and correctly raise errors otherwise.

DanielvanderZee commented:
Requested changes have been incorporated:
- Added functions now try to replace missing values with None and otherwise correctly raise errors;
- Removed @author tags from comments
- Removed unnecessary whitespace.

FlorisHoogenboom commented:
@DanielvanderZee Awesome! Will have a look during lunch!

FlorisHoogenboom committed:
Removed some catch-all clauses. Added some formatting.

FlorisHoogenboom committed:
Renamed method clean_rating to extract_alphanumeric as per recommendation of @tomrijntjes.

FlorisHoogenboom commented:
Hi @DanielvanderZee, I've removed some catch-all clauses in your code. These are generally seen as a bad practice since they make debugging quite hard. Furthermore, I've incorporated the change proposed by @tomrijntjes. 

Everything seems to work. I would be fine with merging. @tomrijntjes do you agree?
